segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/EntryIteratorState.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.tables;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.ObjectBuilder;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import io.pravega.common.util.ArrayView;
+import lombok.Getter;
+import lombok.SneakyThrows;
+
+import java.io.IOException;
+
+/**
+ * Represents the state of a resumable iterator.
+ */
+public class EntryIteratorState {
+    private static final Serializer SERIALIZER = new Serializer();
+
+    private static final int BOOLEAN_BYTES = 1;
+
+    @Getter
+    private final long position;
+
+    @Getter
+    private final boolean reachedEnd;
+
+    @Getter
+    private final boolean shouldClear;
+
+    @Getter
+    private final boolean deletionRecord;
+
+    /**
+     * Creates a new instance of the EntryIteratorState class.
+     *
+     * @param position The position of the TableEntry
+     * @param reachedEnd If the Entry is at the end of the segment (more recently appended).
+     * @param shouldClear Marks if the client should clear their state (provided start position has been truncated).
+     * @param deletionRecord The Entry read is marked for deletion.
+     */
+    EntryIteratorState(long position, boolean reachedEnd, boolean shouldClear, boolean deletionRecord) {
+        Preconditions.checkArgument(isValid(position), "Position must be at least 0 (a non-negative integer).");
+        this.position = position;
+        this.reachedEnd = reachedEnd;
+        this.shouldClear = shouldClear;
+        this.deletionRecord = deletionRecord;
+    }
+
+    boolean isValid(long position) {
+        return position >= 0;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Tip: If all you need `toString` for is for debugging (watching), consider using Lombok's `@ToString` annotation which will auto-generate this for you (you can even pick the fields you want to include).